sunpy/map/mapbase.py

```diff
@@ -166,7 +159,21 @@ class GenericMap(NDData):
     """
 
     def __init__(self, data, header, plot_settings=None, **kwargs):
-
+        """
+        Parameters
+        ----------
+        data : `~numpy.ndarray`, list
+            A 2d list or ndarray containing the map data.
+        header : dict
+            A dictionary of the original image header tags.
+        plot_settings : dict, optional
+            Plot settings.
+
+        Other Parameters
+        ----------------
+        **kwargs :
+            Additional keyword arguments are passed to `NDData` init.
+        """
```

Comment:
I agree with the change from `meta` to `header`, but why to put it inside the `__init__`? All the other information is in the class definition. Using your suggested changes, I don't get the help about the class when doing:
`mapbase.GenericMap?` in ipython or using `help()`

Reply:
Yeah, the parameter documentation should be in the class docstring, although I am struggling to find a reference for this convention.

Reply:
This seems to agree with you: 

>  Use the same sections as outlined above (all except Returns are applicable). The constructor (__init__) should also be documented here, the Parameters section of the docstring details the constructors parameters.

(https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#class-docstring)